# Problem Statement 1:
Blood glucose levels for obese patients have a mean of 100 with a standard deviation of 15.
A researcher thinks that a diet high in raw cornstarch will have a positive effect on
blood glucose levels. A sample of 36 patients who have tried the raw cornstarch diet
have a mean glucose level of 108. Test the hypothesis that the raw cornstarch had an
effect or not.

In [4]:
import numpy as np
import pandas as pd

Let's consider, significance level =5%
so , alpha =(0.05/2) = 0.025
z(1-0.025=0.975) = 1.96

In [5]:
def two_tail_ttest(sam_mean=1,pop_mean=1,sam_size=1,std=1,t=1):
    std_error = std/np.sqrt(sam_size)
    t_test = (sam_mean-pop_mean)/std_error
    if (-t < t_test < t) :
        print("Null Hypothesis is accepted...")
        print("Alternate Hypothesis is rejected...")
    else:
        print("Null Hypothesis is rejected...")
        print("Alternate Hypothesis is accepted...")
    return round(t_test,3)  

In [6]:
two_tail_ttest(sam_mean=108,pop_mean=100,sam_size=36,std=15,t=1.96)

Null Hypothesis is rejected...
Alternate Hypothesis is accepted...


3.2

# Problem Statement 2:
In one state, 52% of the voters are Republicans, and 48% are Democrats. In a second
state, 47% of the voters are Republicans, and 53% are Democrats. Suppose a simple
random sample of 100 voters are surveyed from each state.
What is the probability that the survey will show a greater percentage of Republican
voters in the second state than in the first state?


In [86]:
pr1 = 0.52
pr2 = 0.47
n1 = 100
n2 = 100

#### difference of Standard Deviation in two states for Republicans

In [94]:
def dif_sd(pr1,pr2,n1,n2):
    sd_dif = np.sqrt((pr1*(1-pr1)/n1)+(pr2*(1-pr2)/n2))
    z_diff = (pr2-pr1)/sd_dif
    return z_diff

In [97]:
z_value = dif_sd(pr1,pr2,n1,n2)
z_value

-0.7080278164104213

z_value = -0.708 so, according to z-table,

z(0.2177)= -0.708

Answer, Probability Diff = 0.2177

# Problem Statement 3:
You take the SAT and score 1100. The mean score for the SAT is 1026 and the standard
deviation is 209. How well did you score on the test compared to the average test taker?


In [22]:
x=1100
mean=1026
std=209
z=(x-mean)/std

0.35406698564593303

In [29]:
#According to z- table z(0.354) = 0.63683
z_value = 0.63683*100
print('z_value = ',z_value,'%')

z_value =  63.683 %
